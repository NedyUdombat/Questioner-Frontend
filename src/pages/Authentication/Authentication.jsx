import React from 'react';
import PropTypes from 'prop-types';

// components
import Navbar from '../../components/presentational/Navbar/Navbar';
import Image from '../../components/presentational/Image/Image';
import Register from '../../components/container/Register/Register';
import Login from '../../components/container/Login/Login';

// styles
import './Authentication.scss';

const Authentication = ({ match }) => {
  return (
    <div>
      <Navbar className="nav-dark" />
      <div className="auth">
        <div className="left">
          <Image
            src="https://res.cloudinary.com/nedy123/image/upload/v1558220743/demo/auth-illustration.svg"
            className="wh-100"
            alt="Authentication Illustration"
          />
        </div>
        <div className="right">
          {match.path === '/register' ? (
            <Register {...location} />
          ) : (
            <Login {...location} />
          )}
        </div>
      </div>
    </div>
  );
};

Authentication.propTypes = {
  props: PropTypes.shape({
    match: PropTypes.shape({
      path: PropTypes.string,
    }),
    location: PropTypes.object,
  }),
  match: PropTypes.object,
  location: PropTypes.object,
};

export default Authentication;
