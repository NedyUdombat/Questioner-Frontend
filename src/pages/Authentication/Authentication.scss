@import '../../assets/sass/index';

.auth {
  display: flex;
  padding: 3% 6%;

  .left {
    width: 60%;
  }

  .right {
    width: 40%;
  }
}

@media only screen and (max-width: 47.5rem) {
  .auth {
    display: block;

    .left {
      display: none;
      width: 100%;
    }

    .right {
      width: 100%;
    }
  }
}
