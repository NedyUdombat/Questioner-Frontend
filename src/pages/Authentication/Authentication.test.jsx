import React from 'react';
import { shallow } from 'enzyme/build';
import { BrowserRouter } from 'react-router-dom';

import Authentication from './Authentication';

const propsRegister = {
  match: {
    path: '/register',
  },
};

const propsLogin = {
  match: {
    path: '/login',
  },
};

describe('Authentication', () => {
  it('should render correctly', () => {
    const component = shallow(
      <BrowserRouter>
        <Authentication {...propsRegister} />
      </BrowserRouter>,
    );
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with Login', () => {
    const component = shallow(
      <BrowserRouter>
        <Authentication {...propsLogin} />
      </BrowserRouter>,
    );
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with Login', () => {
    const component = shallow(<Authentication {...propsLogin} />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with register', () => {
    const component = shallow(<Authentication {...propsRegister} />);
    expect(component).toMatchSnapshot();
  });
});
