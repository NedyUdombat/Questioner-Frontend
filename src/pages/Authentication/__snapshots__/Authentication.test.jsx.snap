// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Authentication
    match={
      Object {
        "path": "/register",
      }
    }
  />
</Router>
`;

exports[`Authentication should render correctly with Login 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Authentication
    match={
      Object {
        "path": "/login",
      }
    }
  />
</Router>
`;

exports[`Authentication should render correctly with Login 2`] = `
<div>
  <Navbar
    className="nav-dark"
  />
  <div
    className="auth"
  >
    <div
      className="left"
    >
      <Image
        alt="Authentication Illustration"
        className="wh-100"
        src="https://res.cloudinary.com/nedy123/image/upload/v1558220743/demo/auth-illustration.svg"
      />
    </div>
    <div
      className="right"
    >
      <ConnectFunction />
    </div>
  </div>
</div>
`;

exports[`Authentication should render correctly with register 1`] = `
<div>
  <Navbar
    className="nav-dark"
  />
  <div
    className="auth"
  >
    <div
      className="left"
    >
      <Image
        alt="Authentication Illustration"
        className="wh-100"
        src="https://res.cloudinary.com/nedy123/image/upload/v1558220743/demo/auth-illustration.svg"
      />
    </div>
    <div
      className="right"
    >
      <ConnectFunction />
    </div>
  </div>
</div>
`;
